### PR TITLE
[bugfix] auto-calculate the conditions when reordering the questions in the groups

### PR DESCRIPTION
**Pull request by sieutruc**
Hello,

Scenario : i have 2 questions and 2 groups. The question 2 depending on the options of the question 1. Originally and they are in the same group 1. If i move the question 1 to the group 2, the  "cfieldname" of the table lime_condition is not updated and the relevance of "lime_questions" is not also updated.

Patch: when user moves a question to another group, the code will detect if the old and new groups are different. If yes, limesurvey will try to update the lime_condition table and the "relevance" field in lime_questions table.

Truc

### CODE REVIEW

**Comment by sieutruc** [comment]
Can someone tell me whether this function will be merged to the master branch ? Or do you find out something that is wrong in my patch ?

**Comment by SamMousa** [comment]
Hi sieutruc, 

LS2.06 is currently feature locked. However this seems to be more of a bugfix than a feature in my opinion.

I have some comments that I will add to your changes.

@c-schmitz  Please decide if this is a feature or a bugfix!

File: application/models/Question.php [hunk 1]
```diff
@@ -266,6 +266,16 @@ public function getAdvancedSettingsWithValues($iQuestionID, $sQuestionType, $iSu
             return $aAttributeNames;
         }
 
+        function getQuestionById($qid)
```
Comment (SamMousa) [inline application/models/Question.php]: There is no need for this function.
1. It does not have visibility (yes I know some of our older code doesn't have this as well).
2. Read Yii documentation and you will find that the proper way is: Question::model()->findByPk().

**Comment by Shnoulle** [comment]
Hi, it's not a feature here : it's a Bug fix.

Create a condition (with GUI) . Move the question used in condition : condition is broken. 

@sieutruc : if you can put a bug report, update the commit message to fix this bug. And use Question::model()->findByPk($qid) seems a beter idea.

**Comment by sieutruc** [comment]
@Shnoulle @SamMousa  thank you for your suggestion. :)

I have implemented what you said.

 Can you tell me if it can be merged ?

Truc

**Comment by odanielLNG** [comment]
Hello, 

Do you know in which version, this fix will be available ?

Thanks.

**Comment by Shnoulle** [comment]
Hi @odanielLNG : look at https://github.com/LimeSurvey/LimeSurvey/commit/67df87e841414c668730179ebcbe4d2c6f3055fc it was in 2.06_plus_151109

**Pull request merged**